### PR TITLE
Always use oneshot pbdkf2 in netcoreapp

### PR DESCRIPTION
**Pull request by HaoK**
Part of https://github.com/dotnet/aspnetcore/issues/30941

cc @omajid @vcsjones @GrabYourPitchforks

### CODE REVIEW

File: src/DataProtection/Cryptography.KeyDerivation/src/PBKDF2/Pbkdf2Util.cs [hunk 1]
```diff
@@ -27,16 +28,15 @@ private static IPbkdf2Provider GetPbkdf2Provider()
...
             else
             {
-#if NETSTANDARD2_0 || NET461
                 return new ManagedPbkdf2Provider();
+            }
 #elif NETCOREAPP
-                // fastest implementation on .NET Core for Linux/macOS.
-                // Not supported on .NET Framework
-                return new NetCorePbkdf2Provider();
+            // fastest implementation on .NET Core for Linux/macOS.
```
Comment (omajid) [inline src/DataProtection/Cryptography.KeyDerivation/src/PBKDF2/Pbkdf2Util.cs:34]: Maybe we can remove "for Linux/macOS" if we are also relying on this being the fastest on Windows as well?

**Comment by HaoK** [comment]
@dotnet/aspnet-blazor-eng can someone do a quick review on this?  Its a pretty small change, we are just using the new netcoreapp apis on all platforms rather than special casing mac/linux now that it supports and is faster.

**Review by captainsafia — COMMENTED** [review]
Am I correct in understanding that this will only functionally change the behavior on Windows 7/8?

Where previously they would return `Win8Pbkdf2Provider` or `Win7Pbkdf2Provider` on Windows targets we now only do that if they are not net core apps?

**Comment by HaoK** [comment]
@captainsafia yeah, before we used the old OS specific providers because they were faster than the netcoreapp provider, but with the new api focused on making our scenarios faster, we can just use the netcoreapp provider always when its available

**Comment by GrabYourPitchforks** [comment]
Seems fine to me. Remember: you can also use `#if NET6_OR_GREATER` (or whatever that definition is) to simplify checks when you need to light up on an API that was introduced in a recent .NET version.

**Review by captainsafia — APPROVED** [review]

**Pull request merged**